### ISSUE TITLE
Compute the latency of requests for an individual application instance

### ISSUE THREAD

**Issue by kelapure** [pivotalservices/app-metrics-nozzle#4]
We are looking at the HttpStartStop messageType originating from the gorouter. 
The logmessage that is of interest from the Firehose looks like this ... 

`origin:"gorouter" eventType:HttpStartStop timestamp:1465502702084957735 deployment:"pcfdev" job:"pcfdev" index:"0" ip:"10.0.2.15" httpStartStop:<startTimestamp:1465502702076121968 stopTimestamp:1465502702084924666 requestId:<low:18035419150923496247 high:142127639551750991 > peerType:Client method:GET uri:"http://192.168.11.1/webjars/angularjs/1.2.16/angular-resource.js" remoteAddress:"192.168.11.1:58766" userAgent:"Mozilla/5.0" statusCode:200 contentLength:23837 applicationId:<low:17241914858679868987 high:4711039550123150236 > instanceId:"7f9026fa-ab93-4399-6554-4324afc57e4c" >`

There is currently no way to associate the app InstanceID in the log message  with anything returned by the CC API. The CC API does not provide the instanceId/GUID of the app instances i.e. app-instance-0, app-instance-1.

GET /v2/apps/3b9e245f-3696-47ef-9c0b-e8bb82f96041/instances does not return the InstanceIDs of the app. 3b9e245f-3696-47ef-9c0b-e8bb82f96041 is the GUID of the app itself ... 

```
{"0":{"state":"RUNNING","uptime":21416,"since":1465482960}}
{
   "0": {
      "state": "RUNNING",
      "uptime": 21416,
      "since": 1465482960
   }
}
```
How does one correlate the eventType:HttpStartStop to a particular application ?

**Comment by kelapure** [comment]
Update from @Jim-Campbell

**TL;DR The message already contains the app id, just not how people are used to seeing it in plain text.**

If you look at the snippet below, you'll see applicationId:<low:17241914858679868987 high:4711039550123150236 >.This is the older representation of the App ID, in UUID form. From what I've been able to suss out, this was the original format in Loggregator/CF and at some point changed to the text GUID string, such as 3b9e245f-3696-47ef-9c0b-e8bb82f96041, which is used in log messages, and container metrics.

In order to get /v2/apps info, you'll need to convert the UUID format of two uint64s into the string form of GUID. I wrote a short program to demonstrate the algorithm - it is attached. Sample output:

```
○ → uuid-guid 18035419150923496247 142127639551750991
Lower FA4AAE297A127B37 Upper  1F8F055CC3F334F
guid: 377b127a-29ae-4afa-4f33-3fcc55f0f801
```

I suppose a case could be made for adding the GUID field to the httpstartstop message. At the moment I'm inclined not to advocate for this since so far the tooling people use seems to handle it. But I  could be wrong, and I'll start asking about this in customer contexts. 

[uuidgo.zip](https://github.com/pivotalservices/app-metrics-nozzle/files/391878/uuidgo.zip)
